src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,140 @@
+
+            int affectedRows = statement.executeUpdate();
+
+            if (affectedRows < 1) {
+                throw new RuntimeException("Expected to insert at least one row,"
+                        + " but inserted 0 rows.");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getObject(1, Long.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to insert book into the database: "
+                    + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sqlSelect = "SELECT * FROM books WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sqlSelect)) {
+
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+
+            if (resultSet.next()) {
+                String title = resultSet.getString("title");
+                BigDecimal price = resultSet.getBigDecimal("price");
+
+                Book book = new Book();
+                book.setId(id);
+                book.setTitle(title);
+                book.setPrice(price);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to retrieve book with id: " + id, e);
+        }
+        return null;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> bookList = new ArrayList<>();
+        String sqlSelectAll = "SELECT * from books";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sqlSelectAll)) {
+
+            ResultSet resultSet = statement.executeQuery();
+
+            while (resultSet.next()) {
+                Book book = new Book(
+                        resultSet.getLong("id"),
+                        resultSet.getString("title"),
+                        resultSet.getBigDecimal("price")
+                );
+                bookList.add(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to retrieve all books from the database", e);
+        }
+        return bookList;
+    }
+
+    @Override
+    public Book update(Book book) {
+        String sqlUpdate = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(sqlUpdate)) {
+
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setInt(3, Math.toIntExact(book.getId()));
```

Comment:
Using `Math.toIntExact` can throw an `ArithmeticException` if the `book.getId()` value is too large to fit in an `int`. Consider checking the value or using a `PreparedStatement` method that accepts a `long` directly.